src/webgl/loading.js

```diff
@@ -312,8 +410,24 @@ function parseObj(model, lines) {
                 model.vertexNormals.push(loadedVerts.vn[vertParts[2]].copy());
               }
             }
+            if (usedVerts[vertIndex] && usedVerts[vertIndex]
```

Comment:
Right now I think we just store the *last used material* for each vertex. I think this means that if we reuse the same vertex first with material1, then with material2, then a third time with material1 again, we'll end up with three duplicates of the vertex instead of just two, since we make a new duplicate if the current material is not the same as the last used material.

Maybe instead of storing a map of source index &rarr; material, we can store a map from source index &rarr; another map of material &rarr; destination index. In that world, this if statement would check if `usedVerts[vertIndex][currentMaterial]` is defined yet, and create it if not. (This would maybe have to also absorb the `else` branch right above it maybe.) Then, below, we would add `usedVerts[vertIndex][currentMaterial]` to the current face.

Reply:
Right, I missed out on how I could have optimized the logic here and thus have lesser duplication like in the example you mentioned above. Will work on it now. Thanks for pointing this out. @davepagurek